src/components/Text/src/Text.vue

```diff
@@ -110,34 +110,40 @@ export default {
 			default: 'inherit',
 			validator: (textAlign) => ['inherit', 'left', 'right', 'center'].includes(textAlign),
 		},
+		/**
+		 * Variant - allows four custom font styles through the Theme component
+		 * @values body, title, headline, label
+		 */
+		variant: {
+			type: String,
+			default: 'body',
+			validator: (variant) => ['body', 'title', 'headline', 'label'].includes(variant),
```

Comment:
this seems to hardcode 4 variants into the component, and from looking at the CSS classes below the only customizable properties of each variant are color, fontFamily, and fontWeight.

can you generalize this implementation to allow for arbitrarily many user-defined variants which allow customizing any of the themeable properties of the component? this approach has to be able to do at least that much to reach feature-parity with the current beta release.

Reply:
We only have plans to support four variants and the only thing that can be changed are those three options.

Reply:
we have user-defined variants in beta